### PR TITLE
fix_: bridge tx improvements

### LINKED ISSUE #21180: L1 - L2 transaction fails with `L1_ETH_BRG: Value does not match amount` in blockchain
Looks like this is third party (hopper side) issue. Currently I am unable to perform transaction that includes L1 - L2 bridging neither in develop nor in release builds. 

**Steps:** 

1. Send L1 - L2 transaction. I used eth for testing.
2. Check transaction on blockchain

**Actual result:** Transaction fails with `L1_ETH_BRG: Value does not match amount` 

https://etherscan.io/tx/0x1867537c062312f3e9f19289b6a18c704c76f332a980d6ee2dbc8a74efb28e5b

[Status-debug-logs - 2024-09-04T183851.348.zip](https://github.com/user-attachments/files/16873363/Status-debug-logs.-.2024-09-04T183851.348.zip)

Ethereum Transaction Hash (Txhash) Details | Etherscan 2024-09-04 18-23-04

https://github.com/user-attachments/assets/22f6e03d-b582-431f-bc92-3ade65da32c3

- Status version: nightly, release 2.30
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by saledjenic**
- increase gas estimation factor increased (this one solves reverted tx due to `out of gas` error)
- amount parameter added to the key composition, that makes it unique (this one solves the `Fail with error 'L1_ETH_BRG: Value does not match amount'` error)

Fixes https://github.com/status-im/status-mobile/issues/21180

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]
🚀

File: services/wallet/router/pathprocessor/processor_bridge_hop.go [hunk 1]
```diff
@@ -201,7 +201,7 @@ func (h *HopBridgeProcessor) packTxInputDataInternally(params ProcessorInputPara
 		return []byte{}, createBridgeHopErrorResponse(err)
 	}
 
-	bonderKey := makeKey(params.FromChain.ChainID, params.ToChain.ChainID, "", "")
+	bonderKey := makeKey(params.FromChain.ChainID, params.ToChain.ChainID, "", "", params.AmountIn)
```
Comment (dlipicar) [inline services/wallet/router/pathprocessor/processor_bridge_hop.go:204]: not sure what your original goal is. You want paths from a prior route calculation not to be used when the user changes the amount in the Send/Bridge/Swap popup? If that's the case:
- Shouldn't these tmp values be reset with some other external logic whenever the user changes the input value?
- Why not add uuid to the key to identify paths instead? We should be using a different one on each request to ensure we don't mix results.

Comment (saledjenic) [reply services/wallet/router/pathprocessor/processor_bridge_hop.go:204]: > You want paths from a prior route calculation not to be used when the user changes the amount in the Send/Bridge/Swap popup? 

They are certainly not used cause we do call `Clear()` for each pathprocessor.

> Why not add uuid to the key to identify paths instead? We should be using a different one on each request to ensure we don't mix results.

Results are not mixed cause we clear the state of every pathprocessor on each request.

> Shouldn't these tmp values be reset with some other external logic whenever the user changes the input value?

No, it doesn't need to be reset. Depending on the router input parameters we may have a lot of bridge paths in the candidates list, we need to identify each of them uniquely and that's why the amount is added to the key, but previously that was done using just "from" and to "chain" and that was not enough, cause if the user has positive balances across different chains we're preparing different options for taking certain amount from different chains and the rest from other chain/s and then searching for the best route among them. In the end the best route may have one or more bridge paths, but when sending using just from and to chain doesn't take appropriate bonder data, that's why we've added the amount to the list (key composed of fromChain-toChain-amount is now unique).

**Comment by pavloburykh** [comment]
@saledjenic thanks for the fix! Tested on mobile in https://github.com/status-im/status-mobile/pull/21523 
L1-L2 transactions seem to be fine now

**Examples:**

L1-L2 opt https://etherscan.io/tx/0xd15abe500b8750d80cae89947d4aca9f072ecf06d958f42678ba40a73028063a
L1 - L2 arb https://etherscan.io/tx/0x852de3073514fcbc4889d6322d7b0a6f251d3775ebc32e53bc4e8af69fe02868

**Pull request merged**